snowman/snowman.py

```diff
@@ -14,52 +14,52 @@
 
 
 def snowman(snowman_word):
-    snowman_letters_guessed = build_word_dict(snowman_word)
-    wrong_letters = []
+    correct_letter_guess_statuses = build_letter_status_dict(snowman_word)
+    wrong_guesses_list = []
 
     while True:
-        print(build_game_board(snowman_word, snowman_letters_guessed))
+        print(generate_word_progress_string(snowman_word, correct_letter_guess_statuses))
         user_letter = get_letter_from_user(
-            snowman_letters_guessed, wrong_letters)
+            correct_letter_guess_statuses, wrong_guesses_list)
 
-        if user_letter in snowman_letters_guessed:
+        if user_letter in correct_letter_guess_statuses:
             print(f"Correct! {user_letter} is in the word!")
-            snowman_letters_guessed[user_letter] = True
+            correct_letter_guess_statuses[user_letter] = True
         else:
             print(f"Sorry! {user_letter} isn't in the word!")
-            add_wrong_letter(wrong_letters, user_letter)
+            add_wrong_letter(wrong_guesses_list, user_letter)
 
-        if is_word_guessed(snowman_letters_guessed):
+        if is_word_guessed(correct_letter_guess_statuses):
             # show a winning message along with the full word
             print(f"Congratulations, you win! The word was {snowman_word}")
             return
 
-        print(build_snowman_graphic(len(wrong_letters)))
-        print_wrong_letters(wrong_letters)
-        print_guesses_remaining(wrong_letters)
+        print(build_snowman_graphic(len(wrong_guesses_list)))
+        print_wrong_guesses_list(wrong_guesses_list)
+        print_guesses_remaining(wrong_guesses_list)
 
-        if len(wrong_letters) == SNOWMAN_MAX_WRONG_GUESSES:
+        if len(wrong_guesses_list) == SNOWMAN_MAX_WRONG_GUESSES:
             # show a losing message along with the full word
             print(f"Sorry, you lose! The word was {snowman}")
             return
 
 
-def build_snowman_graphic(num_wrong_guesses):
+def build_snowman_graphic(wrong_guesses_count):
     """This function extracts a portion of the 
     snowman depending on the number of 
     wrong guesses and converts it to a single string
     """
 
     # get the part of the snowman for the number of wrong guesses
     lines = []
-    for line_no in range(num_wrong_guesses - 1):
+    for line_no in range(wrong_guesses_count - 1):
         lines.append(SNOWMAN_IMAGE[line_no])
 
     return "\n".join(lines)
 
 
 # There are no issues in this function
-def get_letter_from_user(word_dict, wrong_letters):
+def get_letter_from_user(correct_letter_guess_statuses, wrong_guesses_list):
```

Comment:
"statuses"  looks so weird to me, like it's not a real word. It's fine, but I needed to confess this. 😆